plugins/renesassimulatoradapter/renesassimulatoradapter_test.go

```diff
@@ -135,13 +135,13 @@ func TestMain(m *testing.M) {
 
 	connection, _, err = websocket.DefaultDialer.Dial("ws://localhost:9000", nil)
 	if err != nil {
-		log.Fatalf("Can't connect to simulator adapter: %s", err)
+		log.Errorf("Can't connect to simulator adapter: %s", err)
 	}
 	defer connection.Close()
 
 	ret := m.Run()
 
-	os.Exit(ret)
+	defer os.Exit(ret)
```

Comment:
Why deffer here?

Reply:
we need defer os.Exit(ret) to support previous defers